### PR TITLE
[exporter/elasticsearch] Update README

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

Correct some errors and add more details.

No functional change, no changelog needed.

### CODE REVIEW

File: exporter/elasticsearchexporter/README.md [hunk 1]
```diff
@@ -25,19 +25,19 @@ This exporter supports sending OpenTelemetry logs to [Elasticsearch](https://www
 - `num_workers` (optional): Number of workers publishing bulk requests concurrently.
 - `index`: The
   [index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices.html)
-  or [datastream](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html)
-  name to publish events to. The default value is `logs-generic-default`. Note: To better differentiate between log indexes and traces indexes, `index` option are deprecated and replaced with below `logs_index`
+  or [data stream](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-streams.html)
+  name to publish events to. The default value is `logs-generic-default`. Note: To better differentiate between log indices and traces indices, `index` option is deprecated and replaced with below `logs_index`
```
Comment (crobert-1) [inline exporter/elasticsearchexporter/README.md:29]:
```suggestion
  name to publish events to. The default value is `logs-generic-default`.
```
Nit: It makes sense to me to make the deprecation more obvious, so I added comments that make sense to me. Feel to ignore, reword, or reformat as you see fit.

File: exporter/elasticsearchexporter/README.md [hunk 2]
```diff
@@ -25,19 +25,19 @@ This exporter supports sending OpenTelemetry logs to [Elasticsearch](https://www
 - `num_workers` (optional): Number of workers publishing bulk requests concurrently.
 - `index`: The
```
Comment (crobert-1) [inline exporter/elasticsearchexporter/README.md:26]:
```suggestion
- `index` (DEPRECATED, please use `logs_index` for logs, `traces_index` for traces): The
```

**Review by crobert-1 — APPROVED** [review]
Thanks for the updates!

**Review by JaredTan95 — APPROVED** [review]

**Review by ycombinator — APPROVED** [review]
Thanks @carsonip!

**Review by codeboten — APPROVED** [review]

**Comment by crobert-1** [comment]
I filed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32286 for the failing unit test on arm, it's unrelated to this change.

**Comment by carsonip** [comment]
@crobert-1 thanks, the tests are passing now

**Comment by JaredTan95** [comment]
close https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32344

**Comment by carsonip** [comment]
> close #32344

I think #32344 is more than a documentation fix which is out of scope of this PR #32221 . I've elaborated in the issue.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**